### ISSUE TITLE
amplify pull & push always add "aws_appsync_dangerously_connect_to_http_endpoint_for_testing": true to aws-exports.js 

### ISSUE THREAD

**Issue by zalanfarkas** [aws-amplify/amplify-cli#9337]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.17.6

### Amplify CLI Version

7.6.4

### What operating system are you using?

Mac

### Amplify Categories

auth, function, api, hosting

### Amplify Commands

pull, push

### Describe the bug

When running `amplify pull` or `amplify push`, they always generate the   aws-exports.js containing "aws_appsync_dangerously_connect_to_http_endpoint_for_testing": true.
Even if we delete the aws-exports.js file, the freshly generated file will contain "aws_appsync_dangerously_connect_to_http_endpoint_for_testing": true.
 It started happening after using `amplify mock` for the first time. However, `amplify mock` is not running when the push or pull is executed. (Also, `amplify mock ` is not reverting the content of  aws-exports.js when terminating, however, that's another glitch.)

### Expected behavior

When running `amplify pull` or `amplify push`, they should generate the   aws-exports.js containing only the cloud-related config.

### Reproduction steps

1. Run `amplify mock`
2. Terminate `amplify mock`
3. Run `amplify push`
4. Run `amplify pull`
5. Assert  "aws_appsync_dangerously_connect_to_http_endpoint_for_testing": true in aws-exports.js

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

Workaround: manually remove "aws_appsync_dangerously_connect_to_http_endpoint_for_testing": true from aws-exports.js before starting the app in dev or building the app locally (e.g. for Ionic app version)

**Comment by marcvberg** [comment]
Closing as duplicate of #9338
